### PR TITLE
Base91 Telemetry Support

### PR DESCRIPTION
**Pull request by TJCoffey**
#16 
Base91 telemetry parser that looks through the comment string for the Base91 format, parses it into a Telemetry object, and returns the object plus the comment string with the telemetry removed. A "telemetry" property was added to both the Position and MIC-E Position objects, and the parsers updated to look for Base91 in the comments.

Added a few basic tests to cover the basic uncompressed, compressed, and MIC-E position use cases. Also ran it against the APRS-IS firehose for a while and it didn't throw any exceptions. I don't think I got any live compressed telemetry that actually uses digital values, though. The majority of traffic seen was from Byonics TinyTrak3 units, which is MIC-E position and altitude, plus 2 analog telemetry values.

### CODE REVIEW

File: lib/Position/Base91ParserUtils.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const Telemetry = require('./../MessageModels/Telemetry.js');
+
+module.exports = {
+    parseBase91Telemetry: (bodyContent) => {
```
Comment (adriann0) [inline lib/Position/Base91ParserUtils.js:6]: Nit, I think we can improve var naming across this function a little bit

- bodyContent -> comment
- msgString -> telemetryMatch

File: lib/Position/Base91ParserUtils.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        //Extact and parse Base91 encoded telemetry from this spec: http://he.fi/doc/aprs-base91-comment-telemetry.txt
+        //This should be a sequence number, 1-5 TM values, and a character pair for 8 binary values (which can only appear if 5 other values are sent)
+        const msgString = bodyContent.match(/(.*)\|([!-{]{4,14})\|(.*)/);
+        if (msgString && msgString[2]) { //Found something. [2] is the TM data itself
+            const comment = bodyContent.replace('|' + msgString[2] + '|',''); //Remove the TM block from the comment. Need to add the pipes back. Regex stripped them.
```
Comment (adriann0) [inline lib/Position/Base91ParserUtils.js:11]: Nit, please add whitespaces after commas separating arguments in function calls

**Review by adriann0 — CHANGES_REQUESTED** [review]
Thanks for the contribution! In general change looks good, I left few comments about improvements that can be made. 

Also, could you please add some tests for digital values and some negative scenarios (with invalid values to make sure function doesn't throw). You can create file Base91ParserUtilsTests.js and put these tests there. You can also add to this file samples provided by documentation.

File: lib/Position/Base91ParserUtils.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const Telemetry = require('./../MessageModels/Telemetry.js');
+
+module.exports = {
+    parseBase91Telemetry: (bodyContent) => {
+        //Extact and parse Base91 encoded telemetry from this spec: http://he.fi/doc/aprs-base91-comment-telemetry.txt
+        //This should be a sequence number, 1-5 TM values, and a character pair for 8 binary values (which can only appear if 5 other values are sent)
+        const msgString = bodyContent.match(/(.*)\|([!-{]{4,14})\|(.*)/);
+        if (msgString && msgString[2]) { //Found something. [2] is the TM data itself
```
[the remark this answers is not in the document]
Comment (TJCoffey) [reply lib/Position/Base91ParserUtils.js]: Good catch that checking msgString[2] was redundant. Added check for even length. The regex checks for the right range (4-14), but not for an even number.

File: lib/Position/Base91ParserUtils.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const Telemetry = require('./../MessageModels/Telemetry.js');
+
+module.exports = {
+    parseBase91Telemetry: (bodyContent) => {
+        //Extact and parse Base91 encoded telemetry from this spec: http://he.fi/doc/aprs-base91-comment-telemetry.txt
+        //This should be a sequence number, 1-5 TM values, and a character pair for 8 binary values (which can only appear if 5 other values are sent)
+        const msgString = bodyContent.match(/(.*)\|([!-{]{4,14})\|(.*)/);
```
[the remark this answers is not in the document]
Comment (TJCoffey) [reply lib/Position/Base91ParserUtils.js]: I was doing a modified version of what aprslib did. Some testing confirmed that the prefix/suffix groups are not needed. I've simplified the regex and made the changes you suggested.

"!-{" is a check for characters in the correct range. APRS Base91 uses characters in the ASCII range 33-123 (! to {) for the encoding. So the regex doubles as a filter for invalid characters in the encoded string.

File: lib/Position/Base91ParserUtils.js (hunk 1 above)
Comment (TJCoffey) [reply lib/Position/Base91ParserUtils.js]: Agreed

File: lib/Position/Base91ParserUtils.js (hunk 2 above)
Comment (TJCoffey) [reply lib/Position/Base91ParserUtils.js]: Fixed

**Review by adriann0 — APPROVED** [review]

**Comment by adriann0** [comment]
@TJCoffey tbh I don't have much experience with Github, I think you should be able to merge but let me know if that's not the case and I'll merge it

**Comment by TJCoffey** [comment]
I've got a message saying that "only those with write access to this repository can merge pull requests." Guess you need to do it.
